databricks/koalas/base.py

```diff
@@ -245,11 +245,17 @@ def __mul__(self, other):
         ):
             return column_op(SF.repeat)(other, self)
 
-        if isinstance(self.spark.data_type, StringType) and (
-            (isinstance(other, IndexOpsMixin) and isinstance(other.spark.data_type, IntegralType))
-            or isinstance(other, int)
-        ):
-            return column_op(SF.repeat)(self, other)
+        if isinstance(self.spark.data_type, StringType):
+            if (
+                isinstance(other, IndexOpsMixin)
+                and isinstance(other.spark.data_type, IntegralType)
+                or isinstance(other, int)
+            ):
```

Comment:
Shall we make sure the range of `and` and `or` when they are mixed?

```suggestion
            if (
                isinstance(other, IndexOpsMixin) and isinstance(other.spark.data_type, IntegralType)
            ) or isinstance(other, int):
```

Reply:
Makes sense, this way makes code clearer! Updated.